### PR TITLE
[NFC] Improve for options_compile.cpp

### PR DESCRIPTION
**Pull request by haonanya**
1. Check return value for consume_front function
2. Pass by reference to avoid copy StringRef

### CODE REVIEW

File: options_compile.cpp [hunk 1]
```diff
@@ -287,10 +287,11 @@ std::string EffectiveOptionsFilter::processOptions(const OpenCLArgList &args,
 
   auto parseClExt = [&](const std::string &clExtStr) {
     llvm::StringRef clExtRef(clExtStr);
-    clExtRef.consume_front("-cl-ext=");
+    if (!clExtRef.consume_front("-cl-ext="))
```
Comment (hewj03) [inline options_compile.cpp:290]: Assert return value is true? The use place of parseClExt requires a starts with "-cl-ext="

Comment (haonanya) [reply options_compile.cpp]: done, thanks.

File: options_compile.cpp [hunk 2]
```diff
@@ -287,10 +287,11 @@ std::string EffectiveOptionsFilter::processOptions(const OpenCLArgList &args,
 
   auto parseClExt = [&](const std::string &clExtStr) {
     llvm::StringRef clExtRef(clExtStr);
-    clExtRef.consume_front("-cl-ext=");
+    assert(clExtRef.consume_front("-cl-ext=") &&
```
Comment (hewj03) [inline options_compile.cpp:290]: "clExtRef.consume_front("-cl-ext=")" needs to be moved out of assert, because assert is noop in release mode

Comment (haonanya) [reply options_compile.cpp]: done, thank you!

File: options_compile.cpp [hunk 3]
```diff
@@ -287,10 +287,13 @@ std::string EffectiveOptionsFilter::processOptions(const OpenCLArgList &args,
 
   auto parseClExt = [&](const std::string &clExtStr) {
     llvm::StringRef clExtRef(clExtStr);
-    clExtRef.consume_front("-cl-ext=");
+    bool startWithGivenPrefix = clExtRef.consume_front("-cl-ext=");
```
Comment (hewj03) [inline options_compile.cpp:290]:
```suggestion
    bool hasPrefix = clExtRef.consume_front("-cl-ext=");
```

Comment (haonanya) [reply options_compile.cpp]: done, thanks.

**Review by hewj03 — APPROVED** [review]
LGTM

**Pull request merged**